internal/server.go

```
@@ -154,20 +254,56 @@ func (s *server) startTCP() {
-		s.stateMachine[req.cmd.Array[1].Bulk] = req.cmd.Array[2].Bulk
-		slog.Info("SM", "state", s.stateMachine)
-		s.mu.Unlock()
+		//Step-1 write the log
+		m, err := json.Marshal(Log{
+			Command: req.cmd,
+			Term:    s.currentTerm,
+		})
+		if err != nil {
+			slog.Error("error while parsing command", "err", err)
+		}
+		s.log.Write(uint64(s.prevLogIndex+1), m)
```

golangci-lint (1.62.2)

Error return value of `s.log.Write` is not checked

(errcheck)